### ISSUE TITLE
[ASNetworkImageNode] imageDidLoad: is called multiple times (Progressive JPEG?)

### ISSUE THREAD

**Issue by CR34L** [facebookarchive/AsyncDisplayKit#2489]
Running multiple ASCollectionNodes in one ViewController with Cells containing ASNetworkImageNodes.

When scrolling the CollectionNode only till the next node the delegate "ASNetworkImageNodeDelegate didLoad" is triggered. And this happens everytime the node will be displayed. 

I thought best praxis is defining the delegate as follows:

```
public func imageNode(_ imageNode: ASNetworkImageNode, didLoad image: UIImage) {
   self.setNeedsLayout();
}

override open func layoutSpecThatFits(_ constrainedSize: ASSizeRange) -> ASLayoutSpec {
        let r = imageRatioFromSize(); // ratio calculation after download
        print("\(image.url): \(r)");

        let imagePlace = ASRatioLayoutSpec(ratio: r, child: imageNode);
        imagePlace.style.spacingAfter = 3.0;
        imagePlace.style.spacingBefore = 3.0;
        return imagePlace;
    }
```

Everytime the node is displayed the PINRemoteImageTask calls callCompletionWithQueue and so the childnode triggers the didload delegate. The image is cached size calculation already done. 

Is this a wanted behaviour? In the **Kittens** Example the ASNetworkImageNodeDelegate is not assigned and the imagesize is defined static. So in this case all is fine.

In my case the ASCollectionView is glitching like the UICollectionNode.

using master branch AsyncDisplayKit (2.0-beta.1), PINCache (3.0.1-beta), PINRemoteImage (3.0.0-beta.5)

**Comment by appleguy** [comment]
@CR34L could you check if the cause of the calls is progressive image loading?  You can disable loading progressive images, or I believe there is a property to check the progressive loading amount as well.

If this is the issue, I can adapt this task into a request to clarify the API.  It should probably have a separate variant, like imageDidLoadToProgress:, while the basic version assumes the client is only interested in 100% loaded.  I am pretty sure today it's called on each load of the image, since technically they are relevant moments in the loading process.

**Comment by CR34L** [comment]
I tried to disable progressive loading and its the same behaviour. I tried with progressive and non-progressive jpegs.

I would say that the didLoad delegate method should only called once - if the image did load as the name says. But with cache enabled the image is loaded everytime the node is displayed. So I understand the naming issue.

I think the default use case is to trigger setNeedsLayout() after the image has been downloaded to calculate sizing, ratio and so on. This should only ne done once and stored to the layout cache.

imageDidLoadToProgress -  sounds nice for progressive data btw.
